### PR TITLE
Fixes: Chatbot examples for custom chatbot + rename `suggestions` -> `examples`

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
- Chatbot examples for custom chatbot
- Rename `suggestions` -> `examples`

Closes: #9471

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @dawoodkhan82, this looks good. However, I still get an error when I click on an example in this demo:

```py
import gradio as gr

def yes_man(message, history):
    if message.endswith("?"):
        return "Yes"
    else:
        return "Ask me anything!"

gr.ChatInterface(
    yes_man,
    chatbot=gr.Chatbot(height=300),
    textbox=gr.Textbox(placeholder="Ask me a yes or no question", container=False, scale=7),
    title="Yes Man",
    description="Ask Yes Man any question",
    theme="soft",
    examples=["Hello", "Am I cool?", "Are tomatoes vegetables?"],
).launch()
```

**Comment by dawoodkhan82** [comment]
@abidlabs Looking into it. It doesn't error out for me though. But it is broken because it doesn't append the user message to the chatbot.

File: gradio/chat_interface.py [hunk 1]
```diff
@@ -582,7 +583,10 @@ def _process_msg_and_trim_history(
 
     def _append_history(self, history, message, first_response=True):
         if self.type == "tuples":
-            history[-1][1] = message  # type: ignore
+            if history:
+                history[-1][1] = message  # type: ignore
+            else:
+                history.append([message, None])
```
Comment (dawoodkhan82) [inline gradio/chat_interface.py:589]: Fixed. This was flipped.

**Review by abidlabs — APPROVED** [review]
Great, everything works for me!

**Comment by abidlabs** [comment]
Will go ahead and merge, since I think getting this in will also solve the issue this user is facing: https://github.com/gradio-app/gradio/issues/9491

**Pull request merged**